src/Proxy.cpp

```diff
@@ -64,9 +64,9 @@ MethodCall Proxy::createMethodCall(const std::string& interfaceName, const std::
     return connection_->createMethodCall(destination_, objectPath_, interfaceName, methodName);
 }
 
-AsyncMethodCall Proxy::createAsyncMethodCall(const std::string& interfaceName, const std::string& methodName)
+AsyncMethodCall Proxy::createAsyncMethodCall(const std::string& interfaceName, const std::string& methodName, uint64_t timeout)
 {
-    return AsyncMethodCall{Proxy::createMethodCall(interfaceName, methodName)};
+    return AsyncMethodCall{Proxy::createMethodCall(interfaceName, methodName), timeout};
 }
 
 MethodReply Proxy::callMethod(const MethodCall& message)
```

Comment:
Per suggestion above, we'd have `MethodReply IProxy::callMethod(const MethodCall& message, uint64_t timeout = 0)`, and this timeout would get forwarded to `message.send(timeout)` method, which could be declared like `MethodReply MethodCall::send(uint64_t timeout = 0) const;`

Similarly for async version of these methods/classes (down at Message level, I think we don't need std::chrono::duration overload, uint64_t timeout is sufficient).